examples/fft_batch_speeds.py

```diff
@@ -1,91 +1,149 @@
 """Fourier Transform speed benchmarks for OAH
 
+.. admonition:: Benchmark your own data
+
+    You can use this script to benchmark your data and find the optimum batch
+    size and FFTFilter. Just load your data in place of the ``input_data_3d``
+    and ``input_data_bg_3d``.
+
+
 This example visualizes the speed for different batch sizes for
-the available FFT Filters. The y-axis shows the average speed of a pipeline
-run for the Off-Axis Hologram class :class:`.OffAxisHologram`, including
-background data processing. Therefore, four FFTs are run per pipeline.
+the available FFT Filters.
+
+In the first graph "Batch Processing Time",
+the y-axis shows the speed - normalised by batch size -
+of an Off-Axis Hologram class :class:`.OffAxisHologram`,
+including background data processing. From this graph, we can conclude that:
+
+   - Optimum batch size is between 32 and 256 for 256x256pix imgs
+     (incl padding), but will be limited by your computer's RAM.
+   - Some Notes:
+      - The batch size is the size of the raw hologram 3D stack in z.
+      - Each pipeline runs 4 FFTs (Data FFT and iFFT + background Data FFT +
+        iFFT). For example, batch size of 8 runs 8*4=32 FFTs.
+      - For CuPy, the data transfer between GPU and CPU is
+        currently very inefficient.
+
+
+In the second graph "FFT Speed for Off-Axis Hologram",
+the y-axis shows the speed of a single FFT.
+From this graph, we can conclude that:
+
+   - Optimum batch size for CuPy and PyFFTW is between 16 and 64 for
+     256x256pix imgs (incl padding). This may change depending on your machine.
+
 
-- Optimum batch size is between 64 and 256 for 256x256pix imgs (incl padding),
-  but will be limited by your computer's RAM.
-- Here, batch size is the size of the 3D stack in z.
-- Note that each pipeline runs 4 FFTs. For example, batch 8 runs 8*4=32 FFTs.
+.. admonition:: Notes on warmup runs and PyFFTW
+
+    Each FFT Filter has a warmup run to reduce outliers.
+    This is especially important for PyFFTW. Always keep a single batch size
+    when using PyFFTW, as during the first run at each batch size,
+    PyFFTW will find the fastest way to perform the FFT.
 
 """
 import time
 import matplotlib.pylab as plt
 import numpy as np
 import qpretrieve
 from qpretrieve.data_array_layout import convert_data_to_3d_array_layout
-from qpretrieve.fourier import FFTFilterNumpy, FFTFilterPyFFTW
+from qpretrieve.fourier import FFTFilterNumpy, FFTFilterPyFFTW, FFTFilterCupy
 
 # load the experimental data
 edata = np.load("./data/hologram_cell.npz")
 
 n_transforms_list = [8, 16, 32, 64, 128, 256]
 subtract_mean = True
 padding = True
-# we take the PyFFTW speeds from the second run
-fft_interfaces = [FFTFilterNumpy, FFTFilterPyFFTW, FFTFilterPyFFTW]
+
+# the first run is always used as a warmup
+fft_interfaces = [FFTFilterNumpy, FFTFilterNumpy]
+if FFTFilterCupy is not None:
```

Comment:
should check against `FFTFilterPyFFTW` here?

Reply:
Was a typo, thanks!